AdobeIms/Test/Unit/Model/ConfigTest.php

```diff
@@ -36,28 +35,129 @@ class ConfigTest extends TestCase
     private $scopeConfigMock;
 
     /**
-     * Prepare test objects.
+     * @var UrlInterface|MockObject
+     */
+    private $urlMock;
+
+    /**
+     * Set up test mock objects
      */
     protected function setUp(): void
     {
-        $this->objectManager = new ObjectManager($this);
         $this->scopeConfigMock = $this->createMock(ScopeConfigInterface::class);
-        $this->config = $this->objectManager->getObject(
-            Config::class,
-            [
-                'scopeConfig' => $this->scopeConfigMock
-            ]
-        );
+        $this->urlMock = $this->createMock(UrlInterface::class);
+
+        $this->config = new Config($this->scopeConfigMock, $this->urlMock);
     }
 
     /**
-     * Get API key test.
+     * Test for \Magento\AdobeIms\Model\Config::getApiKey
      */
     public function testGetApiKey(): void
     {
         $this->scopeConfigMock->expects($this->once())
             ->method('getValue')
-            ->with(self::CONFIG_XML_PATH_API_KEY);
-        $this->config->getApiKey();
+            ->with(Config::XML_PATH_API_KEY)
+            ->willReturn(self::STRING_RETURN);
+
+        $this->assertEquals(self::STRING_RETURN, $this->config->getApiKey());
+    }
+
+    /**
+     * Test for \Magento\AdobeIms\Model\Config::getPrivateKey
+     */
+    public function testGetPrivateKey(): void
+    {
+        $this->scopeConfigMock->expects($this->once())
+            ->method('getValue')
+            ->with(Config::XML_PATH_PRIVATE_KEY)
+            ->willReturn(self::STRING_RETURN);
+
+        $this->assertEquals(self::STRING_RETURN, $this->config->getPrivateKey());
+    }
+
+    /**
+     * Test for \Magento\AdobeIms\Model\Config::getTokenUrl
+     */
+    public function testGetTokenUrl(): void
+    {
+        $this->scopeConfigMock->expects($this->once())
+            ->method('getValue')
+            ->with(Config::XML_PATH_TOKEN_URL)
+            ->willReturn(self::STRING_RETURN);
+
+        $this->assertEquals(self::STRING_RETURN, $this->config->getTokenUrl());
+    }
+
+    /**
+     * Test for \Magento\AdobeIms\Model\Config::getAuthUrl
+     */
+    public function testGetAuthUrl(): void
+    {
+        $this->scopeConfigMock->expects($this->exactly(3))
+            ->method('getValue')
+            ->withConsecutive(
+                [Config::XML_PATH_API_KEY],
+                [Custom::XML_PATH_GENERAL_LOCALE_CODE],
+                [Config::XML_PATH_AUTH_URL_PATTERN]
+            )->willReturn(self::STRING_RETURN);
```

Comment:
If you have more than one call they shouldn't return the same value, otherwise, it's hard to check which value exactly was returned. Also, I don't see any of these returned value been used.

Reply:
It is being used in the string replace, but since we don't use the actual format, nothing is really replaced and the original string is returned.

Reply:
Also, all of them should be returning strings, so the string type is effectively what we are checking, right? And not the difference in actual string values?

Reply:
We have to check the logic. In most cases, the logic is that we get some value and return it back. 
For this method, we have to actually check what method does.

This is an example of how it could look like. You can use `@testWith` annotation or `@dataProvider` method to test for more than one set of values:
```php
        $this->scopeConfigMock->method('getValue')
            ->willReturnMap([
                [Config::XML_PATH_API_KEY, 'ClientIdApiKey'],
                [Custom::XML_PATH_GENERAL_LOCALE_CODE, 'EN'],
                [Config::XML_PATH_AUTH_URL_PATTERN, 'https://example.com/authorize?client_id=#{client_id}&redirect_uri=#{redirect_uri}&locale=#{locale}&scope=openid'],
            ]);

        $this->urlMock->method('getUrl')
            ->willReturn('https://backend.com/adobe_ims/oauth/callback');
        $expectedResult = 'https://example.com/authorize?client_id=ClientIdApiKey&redirect_uri=https://backend.com/adobe_ims/oauth/callback&locale=EN&scope=openid';
        $this->assertEquals($expectedResult, $this->config->getAuthUrl());
```

More about return map:
https://phpunit.de/manual/6.5/en/test-doubles.html#test-doubles.stubs.examples.StubTest5.php